### PR TITLE
fix(telemetry): rename the selector to get studio operation id

### PR DESCRIPTION
**Pull request by bnjjj**
We introduced a new `trace_id` selector format in `1.48.0` which has been misnamed because it's not a trace id but the Apollo Studio Operation ID. If you want to access to this selector, here is an example:

```yaml
telemetry:
  instrumentation:
    spans:
      router:
        "studio.operation.id":
            studio_operation_id: true
```

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [x] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: docs/source/configuration/telemetry/instrumentation/selectors.mdx [hunk 1]
```diff
@@ -32,7 +32,8 @@ The router service is the initial entrypoint for all requests. It is HTTP centri
 
 | Selector           | Defaultable | Values                                    | Description                          |
 |--------------------|-------------|-------------------------------------------|--------------------------------------|
-| `trace_id`         | Yes         | `open_telemetry`\|`datadog`\|`apollo`      | The trace ID                         |
+| `trace_id`         | Yes         | `open_telemetry`\|`datadog`\              | The trace ID                         |
```
Comment (garypen) [inline docs/source/configuration/telemetry/instrumentation/selectors.mdx:35]:
```suggestion
| `trace_id`         | Yes         | `open_telemetry`\|`datadog`              | The trace ID                         |
```

**Review by Geal — APPROVED** [review]

File: docs/source/configuration/telemetry/instrumentation/selectors.mdx (hunk 1 above)
Comment (bnjjj) [reply docs/source/configuration/telemetry/instrumentation/selectors.mdx]: Raaah again !

**Review by garypen — APPROVED** [review]

**Comment by abernix** [comment]
I think tests are failing for a legitimate reason here.

**Review by abernix — CHANGES_REQUESTED** [review]
Tests are failing because the rename isn't quite complete.

**Review by abernix — APPROVED** [review]

**Pull request merged**